packages/puppeteer-extra-plugin-block-resources/index.js

```diff
@@ -91,14 +94,57 @@ class Plugin extends PuppeteerExtraPlugin {
     return this.opts.blockedTypes
   }
 
+  /**
+   * Get the request interception resolution priority.
+   *
+   * Priority for Cooperative Intercept Mode can be configured either through `opts` or by modifying this property.
+   *
+   * @type {number} - A number for the request interception resolution priority.
+   */
+  get interceptResolutionPriority() {
+    return this.opts.interceptResolutionPriority
+  }
+
   /**
    * @private
    */
   onRequest(request) {
     const type = request.resourceType()
     const shouldBlock = this.blockedTypes.has(type)
-    this.debug('onRequest', { type, shouldBlock })
-    return shouldBlock ? request.abort() : request.continue()
+
+    // Cooperative Intercept Mode only available in puppeteer@11+
+    const [currentResolution, currentPriority] = request.interceptResolution
+      ? request.interceptResolution()
+      : []
+
+    // Avoid using resolution `alreay-handled` with typo
+    // Requests are immediately handled in Legacy Mode
+    const alreadyHandled = currentResolution
+      ? currentResolution.endsWith('handled')
+      : true
+
+    this.debug('onRequest', {
+      type,
+      shouldBlock,
+      currentResolution,
+      currentPriority
+    })
+
+    if (alreadyHandled) return
```

Comment:
I think I missed this the first time. Can you walk me through the intent? In cooperative mode, the Puppeteer core will run all handlers and resolve the intercept to the one with the highest priority. Unless you have a special case, your handler should simply do its computation and respond at whatever priority it is set to. If a different response at a higher priority takes place, it will prevail and yours will “lose”. If yours is the highest priority response, it will “win”

Reply:
@benallfree The problem is that if you use a plugin that hasn't been migrated to Cooperative Mode and still uses Legacy Mode, the request is handled right away.

So if we call to `continue`/`abort`/`respond` etc. then we get the `Request is already handled!` error again (I just tried it again to verify). That's why we need this condition here, because we can't asume all plugins are opting-in to the new mode.

Reply:
Oh that’s very smart, I think I’ll suggest this pattern in the official docs! Well done.